lib/rules/no-ex-assign.js

```diff
@@ -13,6 +13,8 @@ const astUtils = require("../util/ast-utils");
 
 module.exports = {
     meta: {
+        type: "problem",
```

Comment:
I think this should be a "suggestion". (Not sure why the category is "Possible Errors", either.)

Reply:
I'm still of the mind that this should be a problem, as it's unlikely that you intentionally meant to overwrite a reference to an exception that is thrown.